internal/tiger/config/api_key.go

```diff
@@ -12,22 +12,40 @@ import (
 
 // Keyring parameters
 const (
-	keyringServiceName     = "tiger-cli"
-	keyringTestServiceName = "tiger-cli-test"
-	keyringUsername        = "api-key"
+	keyringServiceName = "tiger-cli"
+	keyringUsername    = "api-key"
 )
 
+// testServiceNameOverride allows tests to override the service name for isolation
+var testServiceNameOverride string
+
 // GetServiceName returns the appropriate service name for keyring operations
-// Uses a test-specific service name when running in test mode to avoid polluting the real keyring
 func GetServiceName() string {
-	// Use Go's built-in testing detection
+	// Tests should set a unique service name to avoid conflicts
+	if testServiceNameOverride != "" {
+		return testServiceNameOverride
+	}
+
+	// In test mode without an override, panic to catch missing test setup
 	if testing.Testing() {
-		return keyringTestServiceName
+		panic("test must call SetTestServiceName() to set a unique keyring service name")
 	}
 
 	return keyringServiceName
 }
 
+// SetTestServiceName sets a unique service name for testing based on the test name
+// This allows tests to use unique service names to avoid conflicts when running in parallel
+// The cleanup is automatically registered with t.Cleanup()
+func SetTestServiceName(t *testing.T) {
+	testServiceNameOverride = "tiger-test-" + t.Name()
+
+	// Automatically clean up when the test finishes
+	t.Cleanup(func() {
+		testServiceNameOverride = ""
+	})
+}
+
```

Comment:
One potential worry I have with this approach is that there could be race conditions where multiple parallel tests call `SetTestServiceName` concurrently, thereby overwriting the value of `testServiceNameOverride` set by the other tests.

I guess that's not an issue right now, because we don't have any parallel tests within the same package, and tests in different packages are fully isolated from each other (i.e. they don't share memory or this `testServiceNameOverride` variable). Right?

Still, it seems like a potential gotcha for down the road. I feel like it would be better to somehow pass the service name into the calls that access the keyring (even though that would be annoying in many ways).

In any case, I think this is fine for now. Definitely an improvement over what we had 👍

Reply:
Yeah this needs a better refactor but merging for now